OCAExplorer/src/App.tsx

```diff
@@ -14,19 +14,24 @@ import {
 import { CssBaseline } from "@mui/material";
 import { StyledEngineProvider, ThemeProvider } from "@mui/material/styles";
 
-const setNewUserCookie = () => {
-    document.cookie = 'OCAExplorerSeenDemo=true; path=/;';
-}
-const seenDemo: string | undefined = document.cookie.split('; ').find((c) => c.split("=")[0] == 'OCAExplorerSeenDemo');
-
 function App() {
   const [overlayData, setOverlayData] = useState<{
     overlay: OverlayBundle | undefined;
     record: CredentialExchangeRecord | undefined;
   }>({ overlay: undefined, record: undefined });
 
-  const cookieValue = seenDemo
-  console.log(cookieValue);
+  // Track if we should skip the rest of the demo.
+  const [skipDemo, setSkipDemo] = useState<boolean>(localStorage.getItem('OCAExplorerSeenDemo') != null);

```

Comment:
Looks much cleaner. Can we take it one step further and combine the state logic for `skipDemo` and `runDemo`? Skipping the demo is effectively the same as having seen the demo or ending it. Perhaps the state in `localStorage` simply tracks which part of the demo the user has seen.

Reply:
I almost feel like what you had before looked cleaner 😅. It's up to you whether you want to keep the current implementation of storing state in `localStorage` but I have approved it regardless cause I think either approach works.